### PR TITLE
Implement Dynalite back-end server base

### PR DESCRIPTION
**Pull request by DigiPie**
This PR adds the general structure and skeleton code for Dynalite. It is a follow-up to the **PR: Add HTTP nodeJS server and COAP nodeJS server** [#1]. Refer to `DEVELOP.md` for local deployment instructions.

image

**Enhancements 🔢**
This PR,

- combines the execution of the NodeJS-Express server and NodeJS-COAP server into a single app.
- implements PostgreSQL pool query implementation
- receives COAP POST messages via the NodeJS-COAP app at `coap://localhost/` and writes these messages to a local PostgreSQL instance (does not verify yet)
- retrieves these message and displays them as a html page via the NodeJS-HTTP app at `http://localhost:3000/`

**Documentation ✍️**
This PR,

- updates the README.md to provide brief information on the project
- adds a DEPLOY.md which is a local deployment guide

**What's next ⏭**
To implement DTLS to authenticate COAP messages.

### CODE REVIEW

File: src/routes/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  connectionString: process.env.DATABASE_URL
+});
+
+var sql_query = 'SELECT * FROM coap_post';
+
+/* GET home page. */
+router.get('/', function(req, res, next) {
+  pool.query(sql_query, (err, data) => {
+		res.render('index', { 
+			data: data.rows
```
Comment (crazoter) [inline src/routes/index.js]: `data` may be null which will result in failure to display page. Change line 15 to perhaps
`data: (data ? data.rows : [])`
to account for that

File: DEVELOP.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Local development guide
+This guide is made up of two parts, the first being on **web server setup** and the second being about **Raspberry Pi setup**.
+
+## Web server setup
+### Downloads
+
+1. [Install PostgreSQL](https://www.postgresql.org/download/).
+2. [Install NodeJS](https://nodejs.org/en/).
```
Comment (crazoter) [inline DEVELOP.md]: On Linux these can be installed via the command:
`sudo apt-get install postgresql postgresql-contrib nodejs`

`npm install` works the same on Linux of course. 

To run `setup.sql` on Linux you'll need to execute `psql -f setup.sql` instead (see [post on StackOverflow](https://stackoverflow.com/questions/9736085/run-a-postgresql-sql-file-using-command-line-arguments). You may want to execute this command as postgres (see [DigitalOcean tutorial](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-postgresql-on-ubuntu-16-04))).

File: src/routes/index.js (hunk 1 above)
Comment (DigiPie) [reply src/routes/index.js]: Thanks for the suggestion. I have implemented this fix.

image

File: DEVELOP.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+1. Run this project on your local server by using `npm start`.
+2. Stop the server by using `CTRL + C`.
+
+### Testing
+
+1. To test Dynalite without a Raspberry Pi, refer to [COAP-CLI](https://github.com/avency/coap-cli). An example command would be: `coap post coap://localhost/ -p "Hello COAP"`.
+2. Dynalite receives POST COAP messages at `coap://localhost/` which are stored in a PostgreSQL table.
+3. Dynalite displays these stored messages at `http://localhost:3000/`.
+
+## Raspberry Pi setup
```
[the remark this answers is not in the document]
Comment (DigiPie) [reply DEVELOP.md]: Thanks for the suggestion. I've separated the guides.

**Pull request merged**